### PR TITLE
Allow site deletion with IP Rules stored

### PR DESCRIPTION
**Pull request by aerosol**
Forgotten `on_delete` in #3761

### CODE REVIEW

File: priv/repo/migrations/20240220144655_cascade_delete_ip_rules.exs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+defmodule Plausible.Repo.Migrations.CascadeDeleteIpRules do
+  use Ecto.Migration
+
+  def up do
+    drop constraint(:shield_rules_ip, "shield_rules_ip_site_id_fkey")
```
Comment (zoldar) [inline priv/repo/migrations/20240220144655_cascade_delete_ip_rules.exs:5]: `drop_if_exists` perhaps, just to be safe? 😅

File: priv/repo/migrations/20240220144655_cascade_delete_ip_rules.exs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    alter table(:shield_rules_ip) do
+      modify :site_id, references(:sites, on_delete: :delete_all), null: false
+    end
+  end
+
+  def down do
+    drop constraint(:shield_rules_ip, "shield_rules_ip_site_id_fkey")
```
Comment (zoldar) [inline priv/repo/migrations/20240220144655_cascade_delete_ip_rules.exs:13]: ditto about `drop_if_exists`

**Review by zoldar — APPROVED** [review]

File: priv/repo/migrations/20240220144655_cascade_delete_ip_rules.exs (hunk 1 above)
Comment (aerosol) [reply priv/repo/migrations/20240220144655_cascade_delete_ip_rules.exs]: fbf2748e5

File: priv/repo/migrations/20240220144655_cascade_delete_ip_rules.exs (hunk 2 above)
Comment (aerosol) [reply priv/repo/migrations/20240220144655_cascade_delete_ip_rules.exs]: fbf2748e5

**Pull request merged**